sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-files/disbursement-receipt-file-record.ts

```diff
@@ -0,0 +1,47 @@
+import { getDateOnlyFromFormat } from "../../../utilities";
+import { DATE_FORMAT } from "../../models/esdc-integration.model";
+import { DisbursementReceiptRecordType } from "../models/disbursement-receipt-integration.model";
+
+/**
+ * Base class for disbursement receipt record.
+ * All the records of disbursement receipt must be a sub class of this.
+ */
+export class DisbursementReceiptRecord {
+  constructor(
+    protected readonly line: string,
+    protected readonly _lineNumber = 0,
+  ) {}
+
+  public get lineNumber(): number {
+    return this._lineNumber;
+  }
+
+  public get recordType(): DisbursementReceiptRecordType {
+    return this.line.substring(10, 11) as DisbursementReceiptRecordType;
+  }
+
+  /**
+   * Any date record as string is parsed to date object.
+   * @param dateText
+   * @returns Date object.
+   */
+  protected convertToDateRecord(dateText: string) {
+    return getDateOnlyFromFormat(dateText, DATE_FORMAT);
+  }
+
+  /**
+   * Parses any amount field to numeric string by ignoring any junk character is present in data.
+   ** parseFloat is used to convert string like 00520.00 to 520.00.
+   * @param amountText
+   * @returns amount value as numeric string.
+   */
+  protected convertToAmountString(amountText: string) {
```

Comment:
I can understand the point about removing the possible garbage from the decimals (already a known issue) but I would not recommend bypassing a possible NaN in the main part (0, 5). If that value comes to a NaN it should be considered an invalid record and it should be part of the log for the investigation.

Reply:
I get it. 👍

Reply:
The method as it is right now can be just as below, what usually we have been doing in the method itself.
```typescript
+amountText.substring(0, 5)
```
I believe that typeorm will not need the `.toString()` also.
IMO the entire method can be removed.